### PR TITLE
Fix L043 issue when trying to autofix functions

### LINKED ISSUE #2047: Applying rule L043 threw an Exception
## Expected Behaviour
No exception

## Observed Behaviour
```
CRITICAL   [L043] Applying rule L043 threw an Exception:                                                                                                    
Traceback (most recent call last):                                                                                                                          
  File "/Users/tekumara/code/jaffle-shop/.venv/lib/python3.9/site-packages/sqlfluff/core/rules/base.py", line 324, in crawl
    res = self._eval(
  File "/Users/tekumara/code/jaffle-shop/.venv/lib/python3.9/site-packages/sqlfluff/rules/L043.py", line 218, in _eval
    column_reference_segment = next(
StopIteration
```

## Steps to Reproduce
_models/test.sql_
```
SELECT dv_runid
        ,CASE
                WHEN LEAD(dv_startdateutc) OVER (
                                PARTITION BY rowid ORDER BY dv_startdateutc
                                ) IS NULL
                        THEN 1
                ELSE 0
                END loadstate
FROM d
```
## Version
sqlfluff, version 0.8.2
python 3.9

## Configuration

Default out-of-the-box config.

## Notes

This exception isn't thrown by sqlfluff 0.8.1

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

Fixes #2047.  L043 was throwing an error when trying to incorrectly apply the fix for the IS NULL reduction case. This should only be applied when a column is compared to NULL in the condition expression, however, it was attempting to apply to functions. I have added the step to filter out the non-column null comparisons.

### Are there any other side effects of this change that we should be aware of?
No, bugfix

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by jpy-git** [comment]
@barrywhart can you take a look when you get a chance please 😄

File: src/sqlfluff/rules/L043.py [hunk 1]
```diff
@@ -216,11 +216,18 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
...
-                    if segment.type == "column_reference"
+                    (
+                        segment
+                        for segment in condition_expression.segments
+                        if segment.type == "column_reference"
+                    ),
+                    None,
                 )
 
+                # Return None if no column reference is detected (this condition does not apply to funcitons).
```
Comment (barrywhart) [inline src/sqlfluff/rules/L043.py:227]:
```suggestion
                # Return None if no column reference is detected (this condition does not apply to functions).
```

**Review by barrywhart — APPROVED** [review]
Looks great! I noticed a typo in a comment, otherwise 💯.

File: src/sqlfluff/rules/L043.py (hunk 1 above)
Comment (jpy-git) [reply src/sqlfluff/rules/L043.py:227]: actually "funcitons" is the hot new programming paradigm 😛

Comment (jpy-git) [reply src/sqlfluff/rules/L043.py]: fixed it now 😄

**Comment by barrywhart** [comment]
Re: funciton, ever heard this song?

https://www.youtube.com/watch?v=PqiZ_gryYUQ

**Pull request merged**